src/components/forms/Election/index.tsx

```diff
@@ -1,55 +1,48 @@
-import LocationInput from "@/components/input/LocationInput";
-import {Group} from "@mantine/core";
 import Button from "@/components/buttons/Button";
 import ListingCard from "@/components/cards/ListingCard";
-import {CardInterface} from "@/utils/types/card";
+import LocationInput from "@/components/input/LocationInput";
+import { CardInterface } from "@/utils/types/card";
+import { Group } from "@mantine/core";
+import axios from "axios";
+import { useEffect, useRef, useState } from "react";
+
+const Election = ({ theme }: { theme: string }) => {
+    const [restaurants, setRestaurants] = useState<CardInterface[]>([]);
+    const isLoading = useRef(false);
 
-const foodList: CardInterface[] = [
-    {
-        id: "1",
-        restaurantName: "KFC",
-        location: "Kuala Lumpur",
-        image: "https://raw.githubusercontent.com/mantinedev/mantine/master/.demo/images/bg-8.png",
-        cuisine: ["Fast Food"]
-    },
-    {
-        id: "2",
-        restaurantName: "McDonald's",
-        location: "Kuala Lumpur",
-    },
-    {
-        id: "3",
-        restaurantName: "Burger King Kongek Besar Jahanam",
-        image: "https://raw.githubusercontent.com/mantinedev/mantine/master/.demo/images/bg-8.png",
-        cuisine: ["King Kong", "Fast Food"]
-    },
-    {
-        id: "4",
-        restaurantName: "Subweiweiweiweiweiweiweiweiweiweiweiweiweiwei",
-        location: "Kuala Lumpur",
-        image: "https://raw.githubusercontent.com/mantinedev/mantine/master/.demo/images/bg-8.png",
-        cuisine: ["Sandwiches"]
-    },
-    {
-        id: "5",
-        restaurantName: "Texas Chicken",
-        location: "Kuala Lumpur",
-        image: "https://raw.githubusercontent.com/mantinedev/mantine/master/.demo/images/bg-8.png",
-        cuisine: ["Down south"]
-    }
-];
+    useEffect(() => {
+        if (isLoading.current === true) return;
 
-const Election = ({theme}: { theme: string }) => {
+        navigator.geolocation.getCurrentPosition(
```

Comment:
Remove this from useEffect thanks. please attach it to the `Start Voting!` button

Reply:
ideally the pin icon populates the input field and the start voting button solely populates the cards